### PR TITLE
Refactor Apodini Document Export

### PR DESCRIPTION
**Pull request by valentinbootz**
# Refactor Apodini Document Export

## :recycle: Current situation & Problem

[comment]: # "
*Describe the current situation (if possible with and exemplary (or real) code snippet and/or where this is used)*
*Please link any open issue that is addressed with this PR*
"
Currently, ``ApodiniMigratorShared`` provides functionality that is useful to create and export documents in Apodini. This describes a common use case that has now implementations in the ``ApodiniMigration`` target in the Apodini package and also the new ``ApodiniSustainability`` package. 

Other implementations should not depend on the ``ApodiniMigrator`` package to support this use case. Instead, ``ApodiniDocumentExport`` implements a package to share this functionality across these implementations.

## :bulb: Proposed solution

[comment]: # "
*Describe the solution and how this affects the project and internal structure*
"
Refactor ``ApodiniMigrator`` to replace ``ApodiniMigratorShared`` dependency with new ``ApodiniDocumentExport`` package to avoid duplicate code.

## :gear: Release Notes 

[comment]: # "
*Add a short summary of the feature as well as possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented if it appends or changes the public interface.*
"
- Add ApodiniDocumentExport dependency and refactor ApodiniMigratorShared

## :heavy_plus_sign: Additional Information

[comment]: # "
*Provide some additional information if possible*
"

### Related PRs

[comment]: # "
*Reference the related PRs*
"
[#412](https://github.com/Apodini/Apodini/pull/412)

### Testing

[comment]: # "
*Are there tests included? If yes, which situations are tested and which corner cases are missing?*
"
- Testing should not be affected by this PR
- Removes test to encode and decode document in json and yaml

### Reviewer Nudging

[comment]: # "
*Where should the reviewer start, where is a good entry point?*
"
Take a look at the ``ApodiniDocumentExport`` package to see how this PR refactors ``ApodiniMigrator``

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/Apodini/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/Apodini/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/Apodini/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/Apodini/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Sources/ApodiniMigratorShared/EncodeDecode/DecodingStrategies.swift [hunk 1]
```diff
@@ -87,3 +87,33 @@ public extension JSONDecoder {
         return self
     }
 }
+
+extension JSONDecoder.DateDecodingStrategy: Equatable {
```
Comment (PSchmiedmayer) [inline Sources/ApodiniMigratorShared/EncodeDecode/DecodingStrategies.swift]: As far as I can see you have moved the extension from the test target to the main target. Do we need to have them in the main target? If so, would it even make sense to also move these extensions to `ApodiniDocumentExport`? The same also applies to the EncodingStrategies.

**Review by PSchmiedmayer — APPROVED** [review]
Thanks for creating `ApodiniDocumentExport`!
I only had a small comments about the extensions, apart from this it looks good to me 👍

File: Sources/ApodiniMigratorShared/EncodeDecode/DecodingStrategies.swift (hunk 1 above)
Comment (valentinbootz) [reply Sources/ApodiniMigratorShared/EncodeDecode/DecodingStrategies.swift]: Good catch! 💪🏼 Looks to me like we do not need this in the main target so it makes sense to keep this in the test to make it clear.
